FreeRTOS-Plus/Demo/coreHTTP_Windows_Simulator/HTTP_S3_Download_Multithreaded/DemoTasks/S3DownloadMultithreadedHTTPExample.c

```diff
@@ -1132,6 +1127,36 @@ static BaseType_t prvDownloadLoop( void )
             xStatus = pdFAIL;
             break;
         }
+
+        /* Check for the "Connection: close" here (instead of the response
+         * task), so that the failed request can be re-sent */
+        else if( xDownloadRespItem.xResponse.respFlags == HTTP_RESPONSE_CONNECTION_CLOSE_FLAG )
+        {
+            LogInfo( ( "'Connection:close' header found in server response. Attempting to re-connect to the server." ) );
+            vTaskSuspend( xRequestTask );
+            vTaskSuspend( xResponseTask );
+
+            /* Disconnect and re-establish the connection. */
+            TLS_FreeRTOS_Disconnect( &xNetworkContext );
```

Comment:
If there are multiple threads sharing this connection, do we need to synchronize access?

Reply:
Nevermind I see that only task uses the connection.